extensions/amp-analytics/0.1/events.js

```diff
@@ -147,13 +148,45 @@ export class CustomEventTracker extends EventTracker {
     const targetReady =
         this.root.getElement(context, selector, selectionMethod);
 
-    let observers = this.observers_[eventType];
-    if (!observers) {
-      observers = new Observable();
-      this.observers_[eventType] = observers;
+    const isSandboxEvent = startsWith(eventType, 'sandbox-');
+
+    // Push recent events if any.
+    let buffer;
```

Comment:
this is logically identical to your current code, since `buffer` is immutable after initializaiton.